### PR TITLE
xxd: reduce closure size by splitting it into it's own output

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

Before:
```
 ▶ nix path-info -Sh ./result
/nix/store/dkxzrm8ynwfhi4b07aybgqhf8wabjqiz-xxd-vim-9.0.1811      77.9M
```
After:
```
 ▶ nix path-info -Sh ./result-xxd
/nix/store/l9fwjwhlfygxk3njh8mqqzz9h1556psk-vim-9.0.1811-xxd      31.2M
```

when using neovim:
```
command nix store diff-closures /nix/var/nix/profiles/system-117-link/ /nix/var/nix/profiles/system-118-link/
vim: -39900.1 KiB
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by equirosa — APPROVED** [review]
LGTM

**Review by pbsds — APPROVED** [review]

**Comment by Artturin** [comment]
Let's ping the people who are using the vim package for xxd https://github.com/search?q=NOT+is%3Afork+lang%3Anix+%2Fvim.%2Bxxd%2F&type=code

**Comment by SuperSandro2000** [comment]
> Maybe add a stub that errors with a message to use the xxd package

I can't recall that we have prior art of this and worst case it will trick peoples complicated setups into that xxd exists even if it doesn't. So before people start to bikesheed around this: no



> Needs a release note

Sure

**Comment by SuperSandro2000** [comment]
rebased to fix the merge conflict

File: pkgs/applications/editors/vim/common.nix [hunk 1]
```diff
@@ -14,12 +16,21 @@ rec {
...
-      substituteInPlace runtime/ftplugin/man.vim \
-        --replace "/usr/bin/man " "man "
-    '';
+  # Use man from $PATH; escape sequences are still problematic.
+  postPatch = ''
+    substituteInPlace runtime/ftplugin/man.vim \
+      --replace "/usr/bin/man " "man "
+  '';
+
+  # done here instead of postFixup to move manpage not back
```
Comment (bjornfor) [inline pkgs/applications/editors/vim/common.nix:25]: > done here instead of postFixup to move manpage not back

This was difficult for me to read. Reword please?

File: nixos/doc/manual/release-notes/rl-2405.section.md [hunk 2]
```diff
@@ -32,8 +32,7 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 - `mkosi` was updated to v19. Parts of the user interface have changed. Consult the
   [release notes](https://github.com/systemd/mkosi/releases/tag/v19) for a list of changes.
 
-- `services.avahi.nssmdns` got split into `services.avahi.nssmdns4` and `services.avahi.nssmdns6` which enable the mDNS NSS switch for IPv4 and IPv6 respectively.
-  Since most mDNS responders only register IPv4 addresses, most users want to keep the IPv6 support disabled to avoid long timeouts.
+- `xxd` has been moved from `vim` default output to its own output to reduce closure size. The canonical way to reference it across all platforms is `unixtools.xxd`.
```
Comment (bjornfor) [inline nixos/doc/manual/release-notes/rl-2405.section.md:35]: Looks like a bad merge resolution -- this patch shouldn't remove the services.avahi.nssmdns entry.

Comment (SuperSandro2000) [reply nixos/doc/manual/release-notes/rl-2405.section.md:38]: ups, habbit since I was removing that line for a while when the patch was only part of my forked branch.

File: pkgs/applications/editors/vim/default.nix [hunk 3]
```diff
@@ -62,11 +62,4 @@ stdenv.mkDerivation {
   '';
 
   __impureHostDeps = [ "/dev/ptmx" ];
-
-  # To fix the trouble in vim73, that it cannot cross-build with this patch
-  # to bypass a configure script check that cannot be done cross-building.
-  # http://groups.google.com/group/vim_dev/browse_thread/thread/66c02efd1523554b?pli=1
-  # patchPhase = ''
-  #   sed -i -e 's/as_fn_error.*int32.*/:/' src/auto/configure
-  # '';
```
Comment (bjornfor) [inline pkgs/applications/editors/vim/default.nix:71]: Removing this block seems off-topic. Factor out in a separate commit?

Comment (SuperSandro2000) [reply pkgs/applications/editors/vim/default.nix:71]: done

**Review by bjornfor — APPROVED** [review]
LGTM.

**Comment by thiagokokada** [comment]
Branch conflict.

**Review by thiagokokada — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Can we merge this?
